apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -64,77 +66,86 @@ const BuyModal = (props: BuyModalProps) => {
-    `?walletAddress=${defaultAddress}` +
+    `?walletAddress=${defaultAddress.hash}` +
     `&theme=${theme.name}` +
     `&backgroundColor=${theme.bg.primary.slice(1)}` +
     `&textColor=${theme.font.primary.slice(1)}` +
     `&primaryColor=${theme.global.accent.slice(1)}` +
     `&secondaryColor=${theme.global.complementary.slice(1)}`
 
-  const handleModalCloseWhenFinished = (e: WebViewNavigation) => {
+  useEffect(() => {
+    if (!currentUrl) {
+      setCurrentUrl(banxaInitialURL)
+    }
+  }, [defaultAddress, theme, currentUrl, banxaInitialURL])
+
+  if (!defaultAddress) return null
+
+  const handleNavigationChange = (e: WebViewNavigation) => {
     // Close modal if url equals default return URL configured in Banxa dashboard
     if (e.url.includes('alephium.org')) {
       navigation.navigate('ActivityScreen')
-      props.onClose()
+      setCurrentUrl(banxaInitialURL)
+      onClose()
+    } else {
+      setCurrentUrl(e.url)
     }
   }
 
   return (
-    <Portal>
-      <BottomModal
-        title={t('Buy')}
-        maximisedContent
-        noPadding
-        {...props}
-        Content={(props) => (
-          <ModalContent {...props} contentContainerStyle={{ flex: 1, paddingTop: 0 }}>
-            {!isDisclaimerAccepted && (
-              <DisclaimerContent>
-                <ScreenTitle title={t('Disclaimer')} />
-                <AppText style={{ flex: 1 }}>
-                  <Trans
-                    t={t}
-                    i18nKey="banxaDisclaimer"
-                    components={{
-                      1: <LinkToWeb url="https://www.banxa.com" />
-                    }}
-                  >
-                    {
-                      'You are about to access 3rd party services provided by <1>Banxa.com</1> through an in-app browser. Alephium does not control Banxa’s services. Banxa’s terms and conditions will apply, so please read and understand them before proceeding.'
-                    }
-                  </Trans>
-                </AppText>
-                <Button
-                  title={t("Alright, let's get to it.")}
-                  onPress={() => setIsDisclaimerAccepted(true)}
-                  variant="highlight"
-                  style={{ marginBottom: insets.bottom }}
-                />
-              </DisclaimerContent>
-            )}
-            <WebView
-              ref={webViewRef}
-              source={{
-                uri: banxaURL
-              }}
-              allowsInlineMediaPlayback
-              enableApplePay
-              mediaPlaybackRequiresUserAction={false}
-              containerStyle={{ padding: 0 }}
-              allowsBackForwardNavigationGestures
-              onNavigationStateChange={handleModalCloseWhenFinished}
-              nestedScrollEnabled
-            />
-          </ModalContent>
-        )}
-      />
-    </Portal>
+    <BottomModal
+      title={t('Buy')}
+      maximisedContent
+      noPadding
+      onClose={onClose}
```

Comment:
We could replace the `onClose` with `id` and move the `onClose` definition in the `BottomModal`.

Reply:
Is it ok if I do that in a follow up PR?

Reply:
Sure, can you add a TODO in this PR?

Reply:
I've added a note in my [follow up PR](https://github.com/alephium/alephium-frontend/pull/794) :)